src/features/author/author.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Author } from './author.entity';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class AuthorService {
+  constructor(
+    @InjectRepository(Author)
+    private readonly authorRepository: Repository<Author>,
+  ) {}
+
+  async getAuthor() {
+    const count = await this.authorRepository.count();
+    if (count === 0) {
+      return null;
+    }
+
+    const randomOffset = Math.floor(Math.random() * count);
+    const { fullname, id } = await this.authorRepository
+      .createQueryBuilder('author')
+      .skip(randomOffset)
+      .take(1)
+      .getOne();
+
+    return new Promise((resolve) => {
+      setTimeout(() => resolve({ authorId: id, name: fullname }), 5000);
```

Comment:
move 5000 to some constants or a private field in this class

Reply:
Moved to another function